components/ReferralCodeCreator.vue

```diff
@@ -0,0 +1,21 @@
+<script setup lang="ts">
+export interface ReferralCodeCreatorProps {
+  referralsUrl: string
+}
+
+const referralsUrl =  `${process.env.VUE_APP_CONSOLE_URL}/referrals`
```

Comment:
`process.env` will not be available for es modules after build, you should use `import.meta.env` instead where required: https://nuxt.com/docs/api/advanced/import-meta#builder-properties